src/pybind/mgr/cephadm/module.py

```diff
@@ -2318,6 +2338,24 @@ def upgrade_resume(self):
     def upgrade_stop(self):
         return self.upgrade.upgrade_stop()
 
+    @trivial_completion
+    def ok_to_stop(self,
+                   daemon_type: str,
+                   daemon_ids: List[str]):
+        names = [f'{daemon_type}.{d_id}' for d_id in daemon_ids]
+
+        if daemon_type not in ServiceSpec.KNOWN_SERVICE_TYPES:
+            raise orchestrator.OrchestratorError(
+                    f'unknown daemon_type "{daemon_type}"')
+
+        if not self.cephadm_services[daemon_type].ok_to_stop(daemon_ids):
+            raise orchestrator.OrchestratorError(
+                    f'It is NOT safe to stop {names}')
+
+        msg = f'It is presumed safe to stop {names}'
+        self.log.info(msg)
+        return msg
```

Comment:
ok now I see the point. for e.g. NFS etc. Hm.

Reply:
> Hm. do we really need a `ceph orch ok-to-stop $daemon`? I mean, we already have `ceph osd ok-to-stop`.

it's useful if we implement logic like this for NFS, iscsi etc, but it's also useful in case we'd like the extend the OSD etc ok-to-stop checks with Orchestrator specific logic ...